### ISSUE TITLE
Pagination fires deprecation warning by default

### ISSUE THREAD

**Issue by emyarod** [carbon-design-system/carbon#6311]
## What package(s) are you using?

- [x] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Following this change, it seems like it's now impossible to use `<Pagination>` without raising a deprecation warning. If we set the `pageInputDisabled` prop to `true` or `false`, we get the warning. If we set it to `null` or `undefined`, or omit it entirely, it defaults to `false` and we get the warning. This is a problem for us because our unit tests fail if warnings or errors are raised. Any suggestions?

originally reported by @s100 in https://github.com/carbon-design-system/carbon/pull/6070#issuecomment-645946911

**Comment by s100** [comment]
If anybody else is having this problem, at the moment we are working around it by manually deleting the default prop prior to using the component:

```js
import { Pagination } from 'carbon-components-react'

delete Pagination.defaultProps.pageInputDisabled

// then use <Pagination> like normal
```

**Comment by aviral2311** [comment]
Could anyone help me in figuring out which version got this fix?

**Comment by tw15egan** [comment]
Should be included in `7.15.0` 

https://github.com/carbon-design-system/carbon/releases/tag/v10.15.0
